pkg/controller/vsphere/machine_scope.go

```diff
@@ -45,6 +45,12 @@ type machineScope struct {
 	machineToBePatched runtimeclient.Patch
 }
 
+var (
+	formatServerHost = func(serverHost string) string {
```

Comment:
I introduced this function because of an issue I had with the testing environment. 
For our unit tests, we are using a mocked vSphere environment - "simulator", that creates a local vSphere API server. The mocked server hostname is an IP with port - `127.0.0.1:4242` but real vSphere environment should have a resolvable DNS record.
Credentials secret has to contain 2 keys: `serverHost.username` and `serverHost.username` which means it should have  `127.0.0.1:4242.username` and  `127.0.0.1:4242.password` keys.
Creating a credentials secret with `127.0.0.1:4242.username` key is not possible because `:` is not a valid character in the key.
`formatServerHost()` is a "hack" for making tests work. 
If anyone has better ideas, I'm happy to hear them.

Reply:
imo, the hack seems fine but i would add a comment to the code describing why you added it.

Reply:
@JoelSpeed you might be right. I thought vcenter should have a hostname but I checked the docs for UPI - https://docs.openshift.com/container-platform/4.4/installing/installing_vsphere/installing-vsphere.html#installation-vsphere-config-yaml_installing-vsphere  and it seems that IP address is also a valid value

Reply:
This is also true for the vsphere cloud provider https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/k8s-secret.html#create-a-k8s-secret.
We must be compliant with that format.
We should be getting the port from the cloud provider config -> https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html

Reply:
We just need to update `vSphereConfig`  to store the port when calling `getVSphereConfig`, then we honour what ever is given there in our url plumbing or default to 443.